### PR TITLE
Add parallelism to FailureRecovery tests

### PR DESCRIPTION
**Pull request by mwd410**
## Description

This parallelizes the test methods for all FailureRecovery tests. As far as @arhimondr and I could tell, the only other way to do this is using the TestNG XML descriptor, designating the class with `parallel="methods"`, but since we do not use that XML descriptor, this appears to be the only available option. These tests can take upwards of 20, 30, even 50 minutes for some connectors. This suite of tests benefitted greatly from parallelism.

## Additional context and related issues

The primary goal here is to speed up FailureRecoveryTests, improving the iteration cycle time, and preventing CI timeouts. I observed a roughly 40-50% improvement (depending on the connector) with this change.

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/ProgressLoggingListener.java [hunk 1]
```diff
@@ -137,22 +137,7 @@ public void onAfterClass(ITestClass testClass)
             return;
         }
 
-        LOGGER.info("[AFTER CLASS] %s", getName(testClass));
-    }
-
-    private String formatTestName(ITestResult testCase)
```
Comment (arhimondr) [inline testing/trino-testing-services/src/main/java/io/trino/testng/services/ProgressLoggingListener.java:143]: Were these two methods unused?

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/Listeners.java [hunk 2]
```diff
@@ -37,4 +40,24 @@ public static void reportListenerFailure(Class<? extends ITestNGListener> listen
         // Therefore, instead of throwing, we terminate the JVM.
         System.exit(1);
     }
+
+    public static String formatTestName(ITestClass testClass)
```
Comment (arhimondr) [inline testing/trino-testing-services/src/main/java/io/trino/testng/services/Listeners.java:44]: Could you please extract changes to logging into a separate commit?

**Review by arhimondr — COMMENTED** [review]
LGTM % comment

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/ProgressLoggingListener.java (hunk 1 above)
Comment (mwd410) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/ProgressLoggingListener.java:143]: They are used - i moved them to `Listeners`

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/Listeners.java (hunk 2 above)
Comment (mwd410) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/Listeners.java:44]: done

**Review by arhimondr — APPROVED** [review]
